src/test/java/uk/gov/hmcts/reform/sandl/snlrules/rules/sessions/SessionDoesNotHaveAJudgeTests.java

```diff
@@ -57,46 +65,153 @@ public void should_be_no_problem_when_session_does_not_have_a_judge_more_than_4_
 
     @Test
     public void should_be_problem_when_session_does_not_have_a_judge_4_weeks_or_nearer_before_start() {
-        setDateInRules(rules, 2018, 04, 05);
+        setDateInRules(rules, 2018, 4, 1);
 
         rules.insert(new Session(sessionId, null, roomId,
-            OffsetDateTime.of(2018, 4, 10, 9, 0, 0, 0,
+            OffsetDateTime.of(2018, 4, 29, 9, 0, 0, 0,
                 ZoneOffset.UTC),
             Duration.ofMinutes(60), "FTRACK"));
 
         droolsService.clearFactModifications();
         rules.fireAllRules(new RuleNameEqualsAgendaFilter(SESSION_DOES_NOT_HAVE_A_JUDGE_4_WEEKS_OR_LESS_BEFORE_START));
 
-        assertProblems(droolsService,1, 0, 0);
+        assertProblems(droolsService, 1, 0, 0);
 
         droolsService.clearFactModifications();
-        setDateInRules(rules, 2018, 4, 11);
+        setDateInRules(rules, 2018, 4, 30);
         rules.fireAllRules(new RuleNameEqualsAgendaFilter(SESSION_DOES_NOT_HAVE_A_JUDGE_4_WEEKS_OR_LESS_BEFORE_START));
-        assertProblems(droolsService,0, 0, 1);
+        assertProblems(droolsService, 0, 0, 1);
     }
 
     @Test
     public void should_be_problem_when_session_dont_have_a_judge_and_date_changes_to_4_weeks_or_nearer_before_start() {
-        setDateInRules(rules, 2018, 01, 01);
+        setDateInRules(rules, 2018, 1, 1);
 
         rules.insert(new Session(sessionId, null, roomId,
-            OffsetDateTime.of(2018, 4, 10, 9, 0, 0, 0,
+            OffsetDateTime.of(2018, 4, 29, 9, 0, 0, 0,
                 ZoneOffset.UTC),
             Duration.ofMinutes(60), "FTRACK"));
 
         droolsService.clearFactModifications();
         rules.fireAllRules(new RuleNameEqualsAgendaFilter(SESSION_DOES_NOT_HAVE_A_JUDGE_4_WEEKS_OR_LESS_BEFORE_START));
         assertThat(getInsertedProblems(droolsService)).isEmpty();
 
-        setDateInRules(rules, 2018, 04, 01);
+        setDateInRules(rules, 2018, 4, 1);
         droolsService.clearFactModifications();
         rules.fireAllRules(new RuleNameEqualsAgendaFilter(SESSION_DOES_NOT_HAVE_A_JUDGE_4_WEEKS_OR_LESS_BEFORE_START));
 
-        assertProblems(droolsService,1, 0, 0);
+        assertProblems(droolsService, 1, 0, 0);
 
         droolsService.clearFactModifications();
-        setDateInRules(rules, 2018, 4, 11);
+        setDateInRules(rules, 2018, 4, 30);
         rules.fireAllRules(new RuleNameEqualsAgendaFilter(SESSION_DOES_NOT_HAVE_A_JUDGE_4_WEEKS_OR_LESS_BEFORE_START));
-        assertProblems(droolsService,0, 0, 1);
+        assertProblems(droolsService, 0, 0, 1);
+    }
+
+    @Test
```

Comment:
is it possible to write tests that check that severity is different in new problem?

Reply:
+1

Reply:
ok, I've added one going from warning to critical